### PR TITLE
heyefi service: init

### PR DESCRIPTION
**Pull request by ryantm**
Hello.

This is the first time I'm adding a service module to NixOS, so I suspect there are probably plenty of parts to improve. Feedback welcome.

Ryan

### CODE REVIEW

**Comment by jagajaga** [comment]
First of all commit's name should follow guide from `CONTRIBUTING.md`.

File: nixos/modules/services/networking/heyefi.nix [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      };
+
+      uploadDir = mkOption {
+        example = /home/ryantm/picture;
+        description = ''
+          The directory to upload the files to.
+          '';
+      };
+
+      enable = mkOption {
```
Comment (jagajaga) [inline nixos/modules/services/networking/heyefi.nix]: Use `mkEnableOption`. Don't forget to look for examples.

File: nixos/modules/services/networking/heyefi.nix [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        default = false;
+        description = ''
+          Enable heyefi, an Eye-Fi upload server.
+        '';
+      };
+
+      user = mkOption {
+        default = "root";
```
Comment (jagajaga) [inline nixos/modules/services/networking/heyefi.nix]: Better to make additional user (for example call it `heyefi`).
Or this service needs root?

Comment (jagajaga) [inline nixos/modules/services/networking/heyefi.nix]: Don't forget to add ids to https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/misc/ids.nix

File: nixos/modules/services/networking/heyefi.nix [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+      uploadKey = mkOption {
+        default = "";
+        description = ''
+          An Eye-Fi card's upload key.
+          '';
+      };
+
+      uploadDir = mkOption {
+        example = /home/ryantm/picture;
```
Comment (lethalman) [inline nixos/modules/services/networking/heyefi.nix]: Use a string here please, not the straight path as it only exists on your system :) Also please change dir to something more generic, or remove the example at all.

**Comment by lethalman** [comment]
Ok thanks, and welcome!

File: nixos/modules/services/networking/heyefi.nix [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+          User = "${cfg.user}";
+          Restart = "always";
+          ExecStart = "${pkgs.heyefi}/bin/heyefi";
+        };
+
+      };
+
+    environment.etc."heyefi/heyefi.config".text =
+      ''
+        # /etc/heyefi/heyefi.conf: DO NOT EDIT -- this file has been generated automatically.
```
Comment (lethalman) [inline nixos/modules/services/networking/heyefi.nix]: The generated file is readonly, anyway.

**Pull request merged**